gptel.el

```diff
@@ -1118,6 +1131,32 @@ Valid JSON unless NO-JSON is t."
 
 ;;; Saving and restoring state
 
+(defun gptel--restore-props (bounds-alist)
+  "Restore text properties from BOUNDS-ALIST.
+BOUNDS-ALIST is a alist of PROP BOUNDS where BOUNDS is a list of BOUND
+and each BOUND is either (BEG END VAL) or (BEG END).  Except for
+response, the resulting gptel text property value is (PROP . VAL) over
+BEG END.
+
+The legacy (BEG . END) is also supported."
+  (if (symbolp (caar bounds-alist))
+      (mapc
+       (lambda (bounds)
+         (let* ((prop (pop bounds)))
+           (mapc
+            (lambda (bound)
+              (add-text-properties
+               (pop bound) (pop bound)
+               (if (eq prop 'response)
```

Comment:
This supposes that the only two kinds of bounds stored are either `(response (beg end) (beg end)...)` or `(tool (beg end value) (beg end value) ...)`.  This is going to break if we decide to store any other kind of property without a value, like `(ignore (beg end) (beg end)...)`.

Reply:
22af547